docs/data-structures/palindromic-linked-list.mdx

```diff
@@ -0,0 +1,158 @@
+---
+title: Palindromic Linked List 
+slug: /data-structures/palindromic-linked-list
+---
+
+# Introduction
+
+A palindrome is a word or combination of characters that read the same forward and backward.
+Whether a linked list is palindromic or not can be found out using stacks - a data structure. Stacks follow the rule of 'Last In First Out',
+popularly known as 'LIFO'.
+
+# Algorithm
+
+1. Create a linked list
+2. Initialize a stack
+3. Push the data of each node in the linked list into the stack till the end
+4. Initialize a pointer to the head of the linked list
+5. Initialize a temporary variable and store the data present at the top of the stack.
+   Remember that this data is actually the data of the last node of the linked list.
+6. Compare the value in the temporary variable with that of the node pointed to by the pointer in the linked list
+   Remember that the pointer points to the head , that is, the first node of the linke dlist.
+7. If the two values are the same, then move onto the next node in the linked list and pop the current top element in the stack.
+   Repeat step 6 till the end of the stack or linked list is reached or the two values being compared are not the same.
+8. If the end of the linked list is reached, then it means that the linked list is palindromic. Otherwise it is not palindromic.
+
+# Examples
+
+## Example 1
+
+Let the linked list be 1->2->3
+All the data is pushed into the stack in the order: 3 2 1 with 1 at the bottom and 3 at top
+The pointer points towards the head of the linked list that is 1
+It is now compared with the top of the stack which is 3, so they are not equal and hence linked list is not palindrome.
+
+## Example 2
+
+Let the linked list be 1->2->1
+All the data is pushed into the stack 
+Compare the top of the stack(=1) with the data at head of the linked list.
+As they are equal, pop the current top of the stack and pointer points towards the next node(=2)
+Again both are equal. Repeat the above step again. 
+You have reached the end of the linked list. therefore, it is palindromic.
```

Comment:
Therefore should start with a capital letter.

Reply:
Done